<?xml version="1.0"?>
<launch>
  <arg name="robot_ip" default="192.168.1.101"/>

  <!-- Octomap and sensors settings -->
  <param name="move_group/octomap_resolution" type="double" value="0.02" />
  <rosparam ns="/move_group" command="load" file="$(find ur3_moveit)/config/sensors_3d.yaml" />
  <param name="/point_subsample" type="int" value="1" />
  <param name="/move_group/manipulator/longest_valid_segment_fraction" type="double" value="0.005" />

  <!-- Planners -->
  <!-- <param name="move_group/ompl/simplify_solutions" value="true" /> -->

  <!-- duplication !!! See start_sim  -->
  <arg name="xacro_file_name" default="ur3.xacro"/>
  <arg name="xacro_path" default="$(find ur3_description)/urdf/$(arg xacro_file_name)"/>
  <param name="robot_description" command="xacro --inorder '$(arg xacro_path)'" />

  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="filters" value="pointcloud" />
  </include>

  <include file="$(find ur3_moveit)/launch/move_group.launch">
    <arg name="publish_monitored_planning_scene" value="true" />
    <arg name="controllers_definition" value="real_controllers.yaml" />
  </include>

  <arg name="min_payload" default="0.0"/>
  <arg name="max_payload" default="3.0"/>
  <arg name="prefix" default="" />
  <include file="$(find ur_modern_driver)/launch/ur_common.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
    <arg name="min_payload" value="$(arg min_payload)"/>
    <arg name="max_payload" value="$(arg max_payload)"/>
    <arg name="prefix" value="$(arg prefix)" />
  </include>

</launch>
